job-events.proto

```diff
@@ -7,15 +7,31 @@ option java_package = "org.cyverse.events.jobs";
 option java_outer_classname = "JobEvents";
 
 message JobEvent {
-    int32 event_id      = 1;
-    string event_name   = 2;
-    string service_name = 3;
-    string host         = 4; // hostname or ip address
-    string app_id       = 5;
-    string job_id       = 6; // id assigned by the DE
-    string job_state    = 7; // only useful for events related to job status updates.
-    string executor_id  = 8; // id assigned by the execution platform (i.e. HTCondor)
-    string user         = 9; // preferably either email address or username.
-    int64 timestamp     = 10; // should be a milliseconds since the epoch
-    string message      = 11;
+    int32 event_id           = 1;
+    string event_name        = 2;
+    string service_name      = 3;
+    string host              = 4; // hostname or ip address
+    string app_id            = 5;
+    required string job_id   = 6; // id assigned by the DE
+    string job_state         = 7; // only useful for events related to job status updates.
+    string executor_id       = 8; // id assigned by the execution platform (i.e. HTCondor)
+    string user              = 9; // preferably either email address or username.
+    int64 timestamp          = 10; // should be a milliseconds since the epoch
+    string message           = 11;
 }
+
+message StepEvent {
+    int32 event_id           = 1;
+    string event_name        = 2;
+    string service_name      = 3;
+    string host              = 4; // hostname or ip address
+    string app_id            = 5;
+    required string job_id   = 6; // id assigned by the DE
+    string job_state         = 7; // only useful for events related to job status updates.
+    string executor_id       = 8; // id assigned by the execution platform (i.e. HTCondor)
+    string user              = 9; // preferably either email address or username.
```

Comment:
I'm on the fence here, for a couple of reasons.

First, the AMQP protocol allows for a `user_id` to be set, which shows up in the consumed message as a header. So this begs the question, do we need to put the username or user ID in the message?

But, what if we did both? The AMQP header could be the user who produced the message, while the user/userid field could be the actual DE user associated with the actions/events. 

Does this make sense?